### PR TITLE
[71551] OpenVINO 2.0

### PR DESCRIPTION
**Pull request by artyomtugaryov**

### CODE REVIEW

File: model_analyzer/shape_utils.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        node = node.get_node()
+        logging.warning('%s layer of type %s has dynamic output shape.', node.get_friendly_name(), node.get_type_name())
+        return get_shape_safely(partial_shape)
+    return [s for s in partial_shape.to_shape()]
+
+
+def get_shape_safely(partial_shape: PartialShape) -> List[int]:
+    shape = []
+    for i, _ in enumerate(partial_shape):
+        dimension = -1
```
Comment (apaniukov) [inline model_analyzer/shape_utils.py:34]: What if dimension is `1..10`?

File: model_analyzer/model_metadata.py [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+        return not bool(self.model)
+
+    def get_framework(self) -> str:
+        if self.is_onnx:
+            return 'onnx'
+        return self.xml.find('./meta_data/cli_parameters/framework').attrib['value']
+
+    @property
+    def outputs(self) -> List[ConstOutput]:
+        return [model_outputs for model_outputs in self.model.outputs]
```
Comment (apaniukov) [inline model_analyzer/model_metadata.py:90]:
```suggestion
        return self.model.outputs
```

File: model_analyzer/model_metadata.py [hunk 3]
```diff
@@ -0,0 +1,587 @@
...
+
+    @property
+    def inputs(self) -> List[ConstOutput]:
+        return [model_input for model_input in self.model.inputs]
```
Comment (apaniukov) [inline model_analyzer/model_metadata.py:94]:
```suggestion
        return self.model.inputs
```

File: model_analyzer/openvino_core_service.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        self._core = Core()
+
+    @property
+    def core(self) -> Core:
+        return self._core
+
+    def read_model(self, model_path: Path, weights_path: Path) -> Model:
+        return self.core.read_model(str(model_path), str(weights_path))
+
+    def compile_model(self, model: Model, device: str):
```
Comment (apaniukov) [inline model_analyzer/openvino_core_service.py:41]: `-> CompiledModel`

File: model_analyzer/shape_utils.py (hunk 1 above)
Comment (artyomtugaryov) [reply model_analyzer/shape_utils.py:34]: Can you create a ticket? We will support it in the nearest future

**Review by apaniukov — DISMISSED** [review]

File: tests/model_metadata_test.py [hunk 5]
```diff
@@ -45,8 +40,8 @@ def is_obsolete_model_test_params(request) -> Tuple[str, bool]:
 
 
 def test_is_obsolete(is_obsolete_model_test_params: Tuple[str, bool]):
-    model, expected = is_obsolete_model_test_params
-    nmd = NetworkMetaData(*get_network_and_xml_path(model))
+    model_name, expected = is_obsolete_model_test_params
+    nmd = ModelMetaData(*get_xml_and_bin_path(model_name))
```
Comment (apaniukov) [inline tests/model_metadata_test.py:44]: `nmd` -> `mmd`

File: model_analyzer/model_metadata.py [hunk 6]
```diff
@@ -0,0 +1,600 @@
...
+                roles['scores_out'] = result.any_name
+                continue
+            if layout == LayoutTypes.NC:
+                roles['boxes_out'] = result.any_name
+                continue
+            if is_batched_image_layout(layout):  # Layout is NCHW
+                roles['raw_masks_out'] = result.any_name
+        return roles
+
+    def _get_output_roles_for_instance_segm_from_tf(self):
```
Comment (apaniukov) [inline model_analyzer/model_metadata.py:194]: Please add type hints

File: model_analyzer/model_metadata.py [hunk 7]
```diff
@@ -0,0 +1,600 @@
...
+        """Return role predictions for output layers. Instance Segmentation only."""
+        roles = {}
+        framework = self.get_framework()
+        if framework == 'onnx':
+            roles = self._get_output_roles_for_instance_segm_from_onnx()
+        elif framework == 'tf':
+            roles = self._get_output_roles_for_instance_segm_from_tf()
+        return roles or None
+
+    def _get_output_roles_for_instance_segm_from_onnx(self):
```
Comment (apaniukov) [inline model_analyzer/model_metadata.py:174]: Please add type hints

File: model_analyzer/model_metadata.py [hunk 8]
```diff
@@ -0,0 +1,600 @@
...
+            node = result.node
+            layout = parse_node_layout(node)
+            if layout == LayoutTypes.NC:
+                roles['detection_out'] = result.any_name
+                continue
+            if is_batched_image_layout(node.layout):
+                roles['raw_masks_out'] = result.any_name
+        return roles
+
+    def get_yolo_v2_params(self) -> dict:
```
Comment (apaniukov) [inline model_analyzer/model_metadata.py:206]: Inconsistent type hints

File: model_analyzer/model_complexity.py [hunk 9]
```diff
@@ -95,26 +123,26 @@ def get_blob_sizes_and_precisions(layer_provider) -> tuple:
...
                 total_memory_size += layer_provider.get_output_blobs_total_size()
         return total_memory_size
 
     def get_minimum_memory_consumption(self) -> int:
-        input_layer_providers = list(filter(lambda x: x.name in self.input_names, self.layer_providers))
-        all_layer_providers = list(
-            filter(lambda x: not isinstance(x, Constant), self.layer_providers))
-        is_computed = {layer_provider.name: False for layer_provider in all_layer_providers}
+        input__layer_providers = list(filter(lambda x: x.name in self._input_names, self._layer_providers))
+        all__layer_providers = list(
```
Comment (apaniukov) [inline model_analyzer/model_complexity.py:133]: Why `__` inside variable names?

File: model_analyzer/layer_provider.py [hunk 10]
```diff
@@ -71,68 +73,51 @@ def __init__(self, layer: Node):
...
 
     def get_input_shape(self, index: int) -> list:
-        if isinstance(self.layer, Node):
-            partial_shape = self.layer.input(index).get_partial_shape()
-            if partial_shape.is_dynamic:
-                logging.warning('%s layer of type %s has dynamic input shape.', self.name, self.type)
-                return []
-            return list(partial_shape.to_shape())
-        return self.layer.in_data[index].shape
+        return get_shape_for_node_safely(self.layer.input(index))
```
Comment (vgoluben) [inline model_analyzer/layer_provider.py:120]: Inconsistent with identical call in `get_output_shape()`

File: model_analyzer/model_metadata.py [hunk 11]
```diff
@@ -0,0 +1,600 @@
...
+        """Extract model params from the output layer of the model. YOLOv2/TinyYOLOv2 only."""
+        params = {}
+        relevant_attributes = {'classes', 'coords', 'num'}
+        output_attributes = self.outputs[0].node.get_attributes()
+        for attribute in relevant_attributes:
+            params[attribute] = output_attributes.get(attribute)
+
+        return params
+
+    def is_argmax_used(self):
```
Comment (vgoluben) [inline model_analyzer/model_metadata.py:216]: Missing type hint

File: model_analyzer/model_metadata.py [hunk 12]
```diff
@@ -0,0 +1,600 @@
...
+        if len(self.outputs) == 1:
+            output_layer = self.outputs[0]
+            output_shapes = self._get_output_shape(output_layer)
+            if len(output_shapes) < 4:
+                return False
+            reduced_dims = output_shapes[2] == output_shapes[3] == 1
+
+        return 'PRelu' in layers_types and reduced_dims
+
+    def get_layers_ids(self) -> Dict[str, str]:
```
Comment (vgoluben) [inline model_analyzer/model_metadata.py:560]: Inconsistent type hints

**Review by vgoluben — CHANGES_REQUESTED** [review]

**Pull request merged**